File: ht16/example/exception.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

"""."""

# a = 0
# b = 42 / a

try:
    a = 0
    b = 42 / a
except ZeroDivisionError:
    print("No division by zero!")


try:
    a = 0
    b = 42 / a
except Exception:
    print("Some error occured!?")


try:
    a = 0
    b = 42 / a
except ZeroDivisionError:
    print("No division by zero!")
except Exception:
    print("Some error occured!?")


raise Exception("Raising an exception!")

File: ht16/example/exception.php
<?php

$a = 0;
$b = 42 / $a;

function divide($a) {
    if ($a == 0) {
        throw new Exception("Division by zero!");
    }
    return 42 / $a;
}

echo divide(2) . "\n";
//echo divide(0) . "\n";


try {
    echo divide(2) . "\n";
    echo divide(0) . "\n";
} catch (Exception $exception) {
    echo $exception->getMessage();
}
